### PR TITLE
dmarc-metrics-exporter: 0.6.1 -> 0.8.0

### PR DESCRIPTION
**Pull request by Ma27**
###### Description of changes

Also contains an update of uvicorn.
Co-authored by @Arteneko.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Arteneko — COMMENTED** [review]
Apart from the uvicorn pythonOlder check I couldn't test on my server (still a bit too lost on this >>), looks good!

File: pkgs/development/python-modules/uvicorn/default.nix [hunk 1]
```diff
@@ -11,21 +11,24 @@
...
 , websockets
+, hatchling
 , pythonOlder
 }:
 
 buildPythonPackage rec {
   pname = "uvicorn";
-  version = "0.18.2";
+  version = "0.20.0";
   disabled = pythonOlder "3.6";
```
[the remark this answers is not in the document]
Comment (Ma27) [reply pkgs/development/python-modules/uvicorn/default.nix]: Thanks, fixed!

**Comment by Ma27** [comment]
> Apart from the uvicorn pythonOlder check I couldn't test on my server (still a bit too lost on this >>)

Anything we can help you with? :)

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please squash.

File: pkgs/servers/monitoring/prometheus/dmarc-metrics-exporter/default.nix [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
+    owner = "jgosmann";
+    repo = "dmarc-metrics-exporter";
+    rev = "v${version}";
+    hash = "sha256-uZCPEUoWNrcSz5qV24eFdZOOxKkoz+Qn6HpELBwOK2I=";
   };
 
   postPatch = ''
     substituteInPlace pyproject.toml \
-      --replace 'uvicorn = {extras = ["standard"], version = "^0.15.0"}' 'uvicorn = {version = "^0.15.0"}' \
+      --replace 'uvicorn = {extras = ["standard"], version = "^0.19.0"}' 'uvicorn = {version = "^0.19.0"}' \
```
Comment (dotlambda) [inline pkgs/servers/monitoring/prometheus/dmarc-metrics-exporter/default.nix:23]: Add `pythonRelaxDepsHook` to `nativeBuildInputs` and set `pythonRelaxDeps = true` instead.

File: pkgs/servers/monitoring/prometheus/dmarc-metrics-exporter/default.nix [hunk 3]
```diff
@@ -1,28 +1,33 @@
...
+    hash = "sha256-uZCPEUoWNrcSz5qV24eFdZOOxKkoz+Qn6HpELBwOK2I=";
   };
 
+  pythonRelaxDeps = true;
+
   postPatch = ''
     substituteInPlace pyproject.toml \
-      --replace 'uvicorn = {extras = ["standard"], version = "^0.15.0"}' 'uvicorn = {version = "^0.15.0"}' \
-      --replace '"^' '">='
+      --replace 'uvicorn = {extras = ["standard"],' 'uvicorn = {' \
```
Comment (dotlambda) [inline pkgs/servers/monitoring/prometheus/dmarc-metrics-exporter/default.nix:25]: shouldn't be necessary once https://github.com/NixOS/nixpkgs/pull/206834 is merged

Comment (dotlambda) [reply pkgs/servers/monitoring/prometheus/dmarc-metrics-exporter/default.nix:25]: You can remove `postPatch` now.

**Comment by Ma27** [comment]
@dotlambda Done! :)

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 206523` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages marked as broken and skipped:</summary>
  <ul>
    <li>lexicon</li>
    <li>odoo</li>
    <li>python310Packages.hug</li>
    <li>python310Packages.pytest-ansible</li>
    <li>python310Packages.pytest-sanic</li>
    <li>python39Packages.hug</li>
    <li>python39Packages.pytest-ansible</li>
    <li>python39Packages.pytest-sanic</li>
    <li>tumpa</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>timetagger</li>
  </ul>
</details>
<details>
  <summary>223 packages built:</summary>
  <ul>
    <li>ansible (ansible_2_14 ,python310Packages.ansible-core)</li>
    <li>ansible-later (python310Packages.ansible-later)</li>
    <li>ansible-lint (python310Packages.ansible-lint)</li>
    <li>ansible_2_12</li>
    <li>ansible_2_13</li>
    <li>apache-airflow (python310Packages.apache-airflow)</li>
    <li>appthreat-depscan</li>
    <li>atomic-operator</li>
    <li>audible-cli</li>
    <li>baserow</li>
    <li>beets (beetsPackages.beets ,beetsPackages.beets-stable)</li>
    <li>beets-unstable (beetsPackages.beets-unstable)</li>
    <li>cfripper</li>
    <li>crackmapexec</li>
    <li>datasette (python310Packages.datasette)</li>
    <li>dmarc-metrics-exporter</li>
    <li>dyndnsc</li>
    <li>esphome</li>
    <li>etebase-server (python310Packages.etebase-server)</li>
    <li>faraday-cli</li>
    <li>hatch</li>
    <li>home-assistant</li>
    <li>kargo</li>
    <li>lollypop</li>
    <li>mailman</li>
    <li>mailman-hyperkitty</li>
    <li>matrix-synapse</li>
    <li>matrix-synapse-plugins.matrix-synapse-mjolnir-antispam</li>
    <li>matrix-synapse-plugins.matrix-synapse-shared-secret-auth</li>
    <li>mopidy-scrobbler</li>
    <li>mov-cli</li>
    <li>paperless-ngx</li>
    <li>parsedmarc (python310Packages.parsedmarc)</li>
    <li>pgadmin4</li>
    <li>pithos</li>
    <li>platformio</li>
    <li>powerdns-admin</li>
    <li>pylode</li>
    <li>python310Packages.aiocurrencylayer</li>
    <li>python310Packages.aiohttp-apispec</li>
    <li>python310Packages.aiosenz</li>
    <li>python310Packages.amcrest</li>
    <li>python310Packages.ansible</li>
    <li>python310Packages.ansible-compat</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.ansible-runner</li>
    <li>python310Packages.appthreat-vulnerability-db</li>
    <li>python310Packages.aria2p</li>
    <li>python310Packages.asyncwhois</li>
    <li>python310Packages.audible</li>
    <li>python310Packages.authcaptureproxy</li>
    <li>python310Packages.authlib</li>
    <li>python310Packages.axis</li>
    <li>python310Packages.bimmer-connected</li>
    <li>python310Packages.bitcoinrpc</li>
    <li>python310Packages.cirq</li>
    <li>python310Packages.cirq-rigetti</li>
    <li>python310Packages.datasette-publish-fly</li>
    <li>python310Packages.datasette-template-sql</li>
    <li>python310Packages.denonavr</li>
    <li>python310Packages.dependency-injector</li>
    <li>python310Packages.devolo-plc-api</li>
    <li>python310Packages.elastic-apm</li>
    <li>python310Packages.elmax</li>
    <li>python310Packages.elmax-api</li>
    <li>python310Packages.entrance</li>
    <li>python310Packages.entrance-with-router-features</li>
    <li>python310Packages.envoy-reader</li>
    <li>python310Packages.falcon</li>
    <li>python310Packages.fastapi</li>
    <li>python310Packages.fastapi-mail</li>
    <li>python310Packages.geoip2</li>
    <li>python310Packages.gidgethub</li>
    <li>python310Packages.glances-api</li>
    <li>python310Packages.ha-philipsjs</li>
    <li>python310Packages.httpx</li>
    <li>python310Packages.httpx-ntlm</li>
    <li>python310Packages.httpx-socks</li>
    <li>python310Packages.iaqualink</li>
    <li>python310Packages.iotawattpy</li>
    <li>python310Packages.ismartgate</li>
    <li>python310Packages.johnnycanencrypt</li>
    <li>python310Packages.json-logging</li>
    <li>python310Packages.luftdaten</li>
    <li>python310Packages.mocket</li>
    <li>python310Packages.netdata</li>
    <li>python310Packages.onvif-zeep-async</li>
    <li>python310Packages.openapi-core</li>
    <li>python310Packages.ormar</li>
    <li>python310Packages.pontos</li>
    <li>python310Packages.pycfmodel</li>
    <li>python310Packages.pylast</li>
    <li>python310Packages.pypoint</li>
    <li>python310Packages.pypsrp</li>
    <li>python310Packages.pyquil</li>
    <li>python310Packages.pyrmvtransport</li>
    <li>python310Packages.pytelegrambotapi</li>
    <li>python310Packages.pytest-httpx</li>
    <li>python310Packages.qcs-api-client</li>
    <li>python310Packages.requests-aws4auth</li>
    <li>python310Packages.respx</li>
    <li>python310Packages.rsskey</li>
    <li>python310Packages.sanic</li>
    <li>python310Packages.sanic-auth</li>
    <li>python310Packages.sanic-testing</li>
    <li>python310Packages.simple-rest-client</li>
    <li>python310Packages.simple-salesforce</li>
    <li>python310Packages.slowapi</li>
    <li>python310Packages.softlayer</li>
    <li>python310Packages.sopel</li>
    <li>python310Packages.strawberry-graphql</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.teslajsonpy</li>
    <li>python310Packages.timetagger</li>
    <li>python310Packages.total-connect-client</li>
    <li>ttp (python310Packages.ttp)</li>
    <li>python310Packages.uvicorn</li>
    <li>python310Packages.webargs</li>
    <li>python310Packages.webdav4</li>
    <li>python310Packages.whodap</li>
    <li>python310Packages.youtube-search-python</li>
    <li>python310Packages.zeep</li>
    <li>python310Packages.zeversolarlocal</li>
    <li>python39Packages.aiocurrencylayer</li>
    <li>python39Packages.aiohttp-apispec</li>
    <li>python39Packages.aiosenz</li>
    <li>python39Packages.amcrest</li>
    <li>python39Packages.ansible</li>
    <li>python39Packages.ansible-compat</li>
    <li>python39Packages.ansible-core</li>
    <li>python39Packages.ansible-kernel</li>
    <li>python39Packages.ansible-later</li>
    <li>python39Packages.ansible-lint</li>
    <li>python39Packages.ansible-runner</li>
    <li>python39Packages.apache-airflow</li>
    <li>python39Packages.appthreat-vulnerability-db</li>
    <li>python39Packages.aria2p</li>
    <li>python39Packages.asyncwhois</li>
    <li>python39Packages.audible</li>
    <li>python39Packages.authcaptureproxy</li>
    <li>python39Packages.authlib</li>
    <li>python39Packages.axis</li>
    <li>python39Packages.bimmer-connected</li>
    <li>python39Packages.bitcoinrpc</li>
    <li>python39Packages.cirq</li>
    <li>python39Packages.cirq-rigetti</li>
    <li>python39Packages.datasette</li>
    <li>python39Packages.datasette-publish-fly</li>
    <li>python39Packages.datasette-template-sql</li>
    <li>python39Packages.denonavr</li>
    <li>python39Packages.dependency-injector</li>
    <li>python39Packages.devolo-plc-api</li>
    <li>python39Packages.elastic-apm</li>
    <li>python39Packages.elmax</li>
    <li>python39Packages.elmax-api</li>
    <li>python39Packages.entrance</li>
    <li>python39Packages.entrance-with-router-features</li>
    <li>python39Packages.envoy-reader</li>
    <li>python39Packages.etebase-server</li>
    <li>python39Packages.falcon</li>
    <li>python39Packages.fastapi</li>
    <li>python39Packages.fastapi-mail</li>
    <li>python39Packages.geoip2</li>
    <li>python39Packages.gidgethub</li>
    <li>python39Packages.glances-api</li>
    <li>python39Packages.ha-philipsjs</li>
    <li>python39Packages.httpx</li>
    <li>python39Packages.httpx-ntlm</li>
    <li>python39Packages.httpx-socks</li>
    <li>python39Packages.iaqualink</li>
    <li>python39Packages.iotawattpy</li>
    <li>python39Packages.ismartgate</li>
    <li>python39Packages.johnnycanencrypt</li>
    <li>python39Packages.json-logging</li>
    <li>python39Packages.luftdaten</li>
    <li>python39Packages.mocket</li>
    <li>python39Packages.netdata</li>
    <li>python39Packages.onvif-zeep-async</li>
    <li>python39Packages.openapi-core</li>
    <li>python39Packages.ormar</li>
    <li>python39Packages.parsedmarc</li>
    <li>python39Packages.pontos</li>
    <li>python39Packages.pycfmodel</li>
    <li>python39Packages.pylast</li>
    <li>python39Packages.pypoint</li>
    <li>python39Packages.pypsrp</li>
    <li>python39Packages.pyquil</li>
    <li>python39Packages.pyrmvtransport</li>
    <li>python39Packages.pytelegrambotapi</li>
    <li>python39Packages.pytest-httpx</li>
    <li>python39Packages.qcs-api-client</li>
    <li>python39Packages.requests-aws4auth</li>
    <li>python39Packages.respx</li>
    <li>python39Packages.rsskey</li>
    <li>python39Packages.sanic</li>
    <li>python39Packages.sanic-auth</li>
    <li>python39Packages.sanic-testing</li>
    <li>python39Packages.simple-rest-client</li>
    <li>python39Packages.simple-salesforce</li>
    <li>python39Packages.slowapi</li>
    <li>python39Packages.softlayer</li>
    <li>python39Packages.sopel</li>
    <li>python39Packages.strawberry-graphql</li>
    <li>python39Packages.sunpy</li>
    <li>python39Packages.teslajsonpy</li>
    <li>python39Packages.timetagger</li>
    <li>python39Packages.total-connect-client</li>
    <li>python39Packages.ttp</li>
    <li>python39Packages.uvicorn</li>
    <li>python39Packages.webargs</li>
    <li>python39Packages.webdav4</li>
    <li>python39Packages.whodap</li>
    <li>python39Packages.youtube-search-python</li>
    <li>python39Packages.zeep</li>
    <li>python39Packages.zeversolarlocal</li>
    <li>searxng</li>
    <li>soco-cli</li>
    <li>spotdl</li>
    <li>tauon</li>
    <li>theharvester</li>
    <li>tribler</li>
    <li>wapiti</li>
    <li>ytmdl</li>
  </ul>
</details>

**Pull request merged**